core/src/main/scala/io/odin/loggers/AsyncLogger.scala

```diff
@@ -94,11 +99,12 @@ object AsyncLogger {
     * @param maxBufferSize If `maxBufferSize` is set to some value and buffer size grows to that value,
     *                      any new events will be dropped until there is a space in the buffer.
     */
-  def withAsyncUnsafe[F[_]: Timer: ContextShift](
+  def withAsyncUnsafe[F[_]](
       inner: Logger[F],
       timeWindow: FiniteDuration,
       maxBufferSize: Option[Int]
   )(
-      implicit F: ConcurrentEffect[F]
-  ): Logger[F] = F.toIO(withAsync(inner, timeWindow, maxBufferSize).allocated).unsafeRunSync()._1
+      implicit F: Async[F],
+      dispatcher: Dispatcher[F]
```

Comment:
It's not recommended to pass Dispatcher implicitly, you might be better off creating one here and using `allocated` here.

Reply:
In the current version, `odin-core` depends only on `cats-effect-std`. And due to the semantic of the `withAsyncUnsafe` method the `Dispatcher` cannot be instantiated:

```scala
def withAsyncUnsafe[F[_]](
    inner: Logger[F],
    timeWindow: FiniteDuration,
    maxBufferSize: Option[Int]
)(
    implicit F: Async[F]
): Logger[F] = {
  val dispatcher: F[(Dispatcher[F], F[Unit])] = Dispatcher[F].allocated <- still cannot run an effect and access the dispatcher
}
```

Perhaps `odin-core` should depend on the IO as well, then `IORuntime` can be passed as an implicit argument.

This change affects several design decisions. @sergeykolbasov what do you think?

Reply:
I'd say if someone is feeling adventurous enough to deal with `unsafe` API, let them do it on their own by providing a custom dispatcher implicitly. If the users _want_ to deal with unsafety, they should know better how to deal with it on their side.

Reply:
I'd take it explicitly, but I agree about having to manage it on the user's side :)